src/pkg/utils.go

```diff
@@ -105,3 +105,27 @@ func SleepWithContext(ctx context.Context, d time.Duration) error {
 		return ctx.Err()
 	}
 }
+
+func Contains[T comparable](s []T, v []T) bool {
+	// Convert both arrays to maps for efficient lookup
+	map1 := make(map[T]bool)
+	map2 := make(map[T]bool)
```

Comment:
No need to convert `v` to a map, since you only ever iterate it once, so might as well iterate it as a list.